controller/embedding/manager.py

```diff
@@ -134,4 +104,110 @@ def get_embedding_name(
         raise ValueError("attribute not found")
     attribute_name = attribute_item.name
 
-    return f"{attribute_name}-{embedding_type}-{embedding_handle}"
+    name = f"{attribute_name}-{embedding_type}-{platform}"
+
+    if model:
+        name += f"-{model}"
+
+    if apiToken:
+        name += f"-{apiToken[:8]}...{apiToken[-4:]}"
+    
+    return name
+
+
+def recreate_embeddings(project_id: str, embedding_ids: Optional[List[str]] = None) -> None:
+    if not embedding_ids:
+        embeddings = embedding.get_all_embeddings_by_project_id(project_id)
+        if len(embeddings) == 0:
+            return
+        embedding_ids = [str(embed.id) for embed in embeddings]
+    for embedding_id in embedding_ids:
+        embedding.update_embedding_state_waiting(project_id, embedding_id)
+        notification.send_organization_update(
+                project_id,
+                f"embedding:{embedding_id}:progress:{0.0}",
+            )
+        notification.send_organization_update(
+                project_id,
+                f"embedding:{embedding_id}:state:{enums.EmbeddingState.WAITING.value}",
+            )
```

Comment:
![image](https://github.com/code-kern-ai/refinery-gateway/assets/91723236/cdf904fa-2c34-4307-ae88-227acbd7fced)

In the frontend on every WAITING state the embeddings query is refetched. This in turn results in two issues. 
1. At the instant WAITING is sent the changes aren't committed yet. (not very critical since some delay in frontend but still possibly problematic)
2. We refetch n times very quickling

i think after the commit one notification with waiting would be enough (with all for the embedding id part)

Reply:
totally agree, will change the call logic

Reply:
Changed